src/react/hooks/juicebox/useProjectDistributionLimit.ts

```diff
@@ -0,0 +1,35 @@
+import { useContext } from "react";
+import { JBCurrency } from "../../../constants";
+import { useJbFundAccessConstraintsStoreDistributionLimitOf } from "../../generated/hooks";
+import { JBContractContext, useJBContractContext } from "src/react/contexts";
+
+/**
+ * The distribution limit for the project. The maximum amount of funds that can be distributed from the project in a funding cycle.
+ */
+export type DistributionLimit = {
+  /**
+   * The amount of the distribution limit in wei.
+   */
+  amountWei: bigint | undefined,
+  /**
+   * The currency of the distribution limit.
+   */
+  currency: JBCurrency,
+};
+
+/**
+ * Hook to retrieve the distribution limit for the current project.
+ *
+ * @returns The distribution limit for the current project.
+ */
+export function useProjectDistributionLimit(): DistributionLimit {
+  const { contracts: { fundAccessConstraintsStore } } = useJBContractContext();
+  const distributionLimit = useJbFundAccessConstraintsStoreDistributionLimitOf({
+    address: fundAccessConstraintsStore.data,
+  });
+
+  return {
+    amountWei: distributionLimit?.data?.[0],
+    currency: distributionLimit?.data?.[1] as JBCurrency,
```

Comment:
This is a dangerous transformation as it could be undefined

Reply:
Have updated, LMKWYT: 

<img width="593" alt="Screen Shot 2023-11-09 at 1 28 29 pm" src="https://github.com/peeldao/juice-hooks/assets/96150256/25de32a3-757f-4f74-9299-47816a1b631b">

Reply:
or `as JBCurrency | undefined` ?